crates/top/re_sdk/src/recording_stream.rs

```diff
@@ -1013,11 +1015,37 @@ impl RecordingStream {
 
         let components: Result<Vec<_>, ChunkError> = components
             .into_iter()
-            .map(|batch| Ok((batch.name(), batch.to_arrow_list_array()?)))
+            .map(|batch| {
+                Ok((
+                    batch.descriptor().into_owned(),
+                    batch.to_arrow_list_array()?,
+                ))
+            })
             .collect();
 
-        let components: IntMap<ComponentName, ArrowListArray<i32>> =
-            components?.into_iter().collect();
+        let components: ChunkComponents = components?.into_iter().collect();
+
+        {
```

Comment:
makes sense but how did this validation end up in here? isn't that more general than send_columns?

Reply:
oO I did not write this code, I'm not sure what's going on